Group all 'New XXX...' operations in a 'New...' sub menu in the file system dock context menu

Maran23 opened the pull request:
This PR improves the context menu in the file system dock by grouping all `New XXX...` operations into a `New...` sub menu.
This will reduce the height of the rather large context menu and gives us the possibility to add more `New XXX...` operations in the future without cluttering the menu and thus making it taller.

| Before | After |
| - | - |
| ![image](https://user-images.githubusercontent.com/66004280/208479122-9c232af2-6113-4e2e-80ea-f76d0f8c2b3f.png) | ![image](https://user-images.githubusercontent.com/66004280/209286893-9f6daa43-08ee-463c-b52a-076465b99423.png) |

Maran23 committed:
Grouped all 'New XXX...' operations in a 'New...' sub menu

This will reduce the height of the rather large context menu and gives us the possibility to add more 'New XXX...' operations in the future without cluttering the menu and thus making it taller.

MarcusElg commented:
Is the "New" prefix really needed in the submenu? Imo it's enough with Folder/Scene/Scripts etc..

So Folder... instead of New Folder...

Maran23 commented:
> Is the "New" prefix really needed in the submenu? Imo it's enough with Folder/Scene/Scripts etc..
> 
> So Folder... instead of New Folder...

Sure. Just had a look and this is the way Windows and IntelliJ are doing it, so sounds reasonable.
|Windows (German)|IntelliJ|
|-|-|
| ![image](https://user-images.githubusercontent.com/66004280/208497281-5fb48ef5-2b98-49ba-bae5-cfaf5d4d0846.png) | ![image](https://user-images.githubusercontent.com/66004280/208497351-c19292d0-7842-41b1-b50f-ad236eca2009.png) |

akien-mga commented:
> Why is the new thing being created ellipsized? Eg. `New... -> Folder...` instead of `New... -> Folder`.

The ellipsis is a convention for "opens a dialog". So it should actually be `New -> Folder...`.

Maran23 commented:
> The ellipsis is a convention for "opens a dialog". So it should actually be `New -> Folder...`.

Changed! I now just removed the ellipsis `...` -> `New.`
I have no strong opinion here, both ideas are nice and understandable. 😄

Kemeros commented:
Shouldn't New be at the top?

Maran23 commented:
> Shouldn't New be at the top?

Maybe, but we should probably discuss this with others in an godot proposal instead.
In this PR, I just grouped them together. My goal was not to move the `New` item to the top.

KoBeWi commented:
It shouldn't. Right now it matches the "New" position on Windows (and likely other systems/file managers).
